services/brig/src/Brig/API/Internal.hs

```diff
@@ -171,6 +176,42 @@ authAPI =
     :<|> Named @"login-code" getLoginCode
     :<|> Named @"reauthenticate" reauthenticate
 
+federationRemotesAPI :: ServerT BrigIRoutes.FederationRemotesAPI (Handler r)
+federationRemotesAPI =
+  Named @"get-federation-remotes" getFederationRemotes -- TODO: get this from TVar!
+    :<|> Named @"add-federation-remotes" addFederationRemote
+    :<|> Named @"delete-federation-remotes" deleteFederationRemotes
+
+addFederationRemote :: FederationDomainConfig -> ExceptT Brig.API.Error.Error (AppT r) ()
+addFederationRemote fedDomConf = do
+  result <- lift . wrapClient $ Data.addFederationRemote fedDomConf
+  case result of
+    Data.AddFederationRemoteSuccess -> pure ()
+    Data.AddFederationRemoteMaxRemotesReached ->
+      throwError . fedError . FederationUnexpectedError $
+        "Maximum number of remote backends reached.  If you need to create more connections, \
+        \please contact wire.com."
+
+getFederationRemotes :: ExceptT Brig.API.Error.Error (AppT r) FederationDomainConfigs
+getFederationRemotes = lift $ do
+  db <- wrapClient Data.getFederationRemotes
+  cfg <- asks (fromMaybe [] . setFederationDomainConfigs . view settings)
+  -- FUTUREWORK: we should solely rely on `db` in the future; `cfg` is just for an easier,
```

Comment:
@fisx This note ties into my comment above. I don't think that a tvar/ioref is needed when calling the GET API, as a DB call for the latest info would be ideal for other services to be up to date.

Reply:
no, this is about adding the old remotes considered in the config file to the response for a transition period, until the operators have had a chance to load the cassandra table.

i think this should stay.  if you still disagree let's call.